android/src/main/java/com/oney/WebRTCModule/DailyWebRTCDevicesManager.java

```diff
@@ -0,0 +1,201 @@
+        if (camera2supported) {
+            Log.d(TAG, "Creating video capturer using Camera2 API.");
+            cameraEnumerator = new Camera2Enumerator(this.reactContext);
+        } else {
+            Log.d(TAG, "Creating video capturer using Camera1 API.");
+            cameraEnumerator = new Camera1Enumerator(false);
+        }
+        return cameraEnumerator;
+    }
+
+    ReadableArray enumerateDevices() {
+        WritableArray devicesArray = Arguments.createArray();
+        this.fillVideoInputDevices(devicesArray);
+        this.fillAudioDevices(devicesArray);
+        return devicesArray;
+    }
+
+    private void fillVideoInputDevices(WritableArray enumerateDevicesArray){
+        String[] devices = cameraEnumerator.getDeviceNames();
+        for (int i = 0; i < devices.length; ++i) {
+            String deviceName = devices[i];
+            boolean isFrontFacing;
+            try {
+                // This can throw an exception when using the Camera 1 API.
+                isFrontFacing = cameraEnumerator.isFrontFacing(deviceName);
+            } catch (Exception e) {
+                Log.e(TAG, "Failed to check the facing mode of camera");
+                continue;
+            }
+            String label = isFrontFacing ? "Front camera" : "Rear camera";
+            String deviceID = isFrontFacing ? "CAMERA_USER" : "CAMERA_ENVIRONMENT";
+            WritableMap params = this.createWritableMap(deviceID, label, DeviceKind.VIDEO_INPUT.getKind());
+            params.putString("facing", isFrontFacing ? "user" : "environment");
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private void fillAudioDevices(WritableArray enumerateDevicesArray){
+        AudioDeviceInfo[] audioOutputDevices = this.audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS);
+
+        boolean isWiredHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(
+                device -> device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADSET ||
+                        device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADPHONES
+        );
+
+        WritableMap params = isWiredHeadsetPlugged ?
+                this.createWritableMap(AudioDeviceType.WIRED_OR_EARPIECE.toString(), "Wired headset", DeviceKind.AUDIO.getKind()) :
+                this.createWritableMap(AudioDeviceType.WIRED_OR_EARPIECE.toString(), "Phone Earpiece", DeviceKind.AUDIO.getKind());
+        enumerateDevicesArray.pushMap(params);
+
+        //speaker
+        params = this.createWritableMap(AudioDeviceType.SPEAKER.toString(), "Speaker", DeviceKind.AUDIO.getKind());
+        enumerateDevicesArray.pushMap(params);
+
+        boolean isBluetoothHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(device -> device.getType() == AudioDeviceInfo.TYPE_BLUETOOTH_SCO);
+        if(isBluetoothHeadsetPlugged){
+            params = this.createWritableMap(AudioDeviceType.BLUETOOTH.toString(), "Bluetooth", DeviceKind.AUDIO.getKind());
```

Comment:
to be consistent with the string "Wired headset", should this be called "Bluetooth headset"?

Reply:
In this case we could connect any bluetooth device, and It would work. So, I think bluetooth would be better here.

Reply:
I'm convinced—let's leave the naming patterns divergent here, since Bluetooth speakers are relatively common but wired speakers are super rare. So we'll have:
* "Bluetooth"
* "Wired headset"